vllm/config.py

```diff
@@ -2386,6 +2381,21 @@ def init_during_runtime(self):
                 self.inductor_compile_sizes = []
             self.compile_sizes = self.inductor_compile_sizes
 
+        # sort to make sure cudagraph capture sizes are in descending order
+        self.capture_sizes.sort(reverse=True)
+
+
+_BATCH_SIZE_ALIGNMENT = 8
+# all the token sizes that **can** be captured by cudagraph.
+# they can be arbitrarily large.
+# currently it includes: 1, 2, 4, 8, 16, 24, 32, 40, ..., 8192.
+# the actual sizes to capture will be determined by the model,
+# depending on the model's max_num_seqs.
+# NOTE: get_graph_batch_size needs to be updated if this list is changed.
+_BATCH_SIZES_TO_CAPTURE = [1, 2, 4] + [
+    _BATCH_SIZE_ALIGNMENT * i for i in range(1, 1025)
+]
```

Comment:
Shouldn't this be tunable? At least the max batch size should be tunable I believe.

Reply:
Also, how does it work with V1? V1 uses slightly different batch sizes for CUDA graphs.

Reply:
these are all possibly sizes to capture, and you can see the max is very large, i.e. `8192` .

later, the actual max batchsize will be determined by `max_num_seqs` .

v1 does not use these fields.

Reply:
Got it. Thanks for the explanation.